### PR TITLE
Promote Expandables from molecule to organism

### PR DESCRIPTION
**Pull request by Scotchester**
The Design Ops team has been given blessing by @schaferjh to get Wells inside of Expandables, one way or another, before our Auto Loans project launches next Wednesday. In discussion with @anselmbradford, we decided that the most "atomic" way to do this was to promote Expandables to an organism, so that they could contain other organisms (e.g., Wells). (The other option would be to demote Wells to a molecule, but there are ostensibly plans for Wells to be more complicated in the future.)

This PR makes that happen.

## Additions

- Ability to use Wells inside Expandables 

## Changes

- Everything about Expandable that is needed to convert it from a molecule to an organism.

## Testing

1. Pull branch
2. `./cfgov/manage.py migrate`
3. `gulp styles && gulp scripts`
4. `./cfgov/manage.py runserver`
5. Look at any page that has expandables and verify that they are still correctly functioning.
6. Log in to Wagtail and create a test page (Browse basic) with an Expandable Group, and put a Well inside one of its Expandables.
7. Test that page out, verify everything works and that there are no JS errors.

## Review

- @jimmynotjim 
- @KimberlyMunoz 
- @anselmbradford 
- @sebworks 
- @kurtw 

## Screenshots

screen shot 2016-05-31 at 11 26 09

## Notes

~~**One major question:**

#2055 set up atomic initialization flags. The init functions were all set to return undefined if they had already been initialized. After I converted Expandables to an organism, their inits started happening _before_ ExpandableGroups, and a JS error was being thrown when ExpandableGroups went to init. The error would prevent the accordion-style groups from only keeping one open at a time.

Because I couldn't figure out how to change the order of the initializations so that ExpandableGroups would keep being initialized first, I "solved" this by, on a hunch, setting the Expandable init function to `return this;` if it found it had already been initialized. This returned the correct Expandable object to the ExpandableGroup init and let it proceed.

If this is kosher, the `return this;` should, I think, be propagated to all of the atomic init functions? If not, I need advice on how to change the order of the initializations so that ExpandableGroups go before Expandables.

Unfortunately, @anselmbradford is out this week, and this PR is pretty time sensitive. @jimmynotjim and @KimberlyMunoz, what do you think?~~

**Update:** Solved with #2171!

## Todos

- Confirm with @duelj and @ajbush whether the padding at the bottom of a well is supposed to be 30px (what the written spec says) or 60px (what the image on the spec page shows)

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

**Comment by jimmynotjim** [comment]
I'm getting some style issues in the beta banner. It's not currently in use, but could be for special announcements 

screen shot 2016-06-01 at 1 53 31 pm

**Comment by Scotchester** [comment]
I noticed that, too, but was hoping it was unrelated to my changes. I'll see if I can debug it.

File: cfgov/jinja2/v1/_includes/organisms/header.html [hunk 1]
```diff
@@ -22,30 +22,30 @@
...
             <div class="m-global-banner_head">
                 <span class="cf-icon
                              cf-icon-error-round
                              m-global-banner_icon"></span>
                 This beta site is a work in progress.
             </div>
-            <div class="m-expandable_content">
-                <p class="m-global-banner_desc  m-expandable_content-animated">
+            <div class="o-expandable_content">
+                <p class="m-global-banner_desc  o-expandable_content-animated">
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/organisms/header.html]: Update with `o-expandable_content-animated` as a wrapper around `m-global-banner_desc`. Mixing two atomic components, especially two different types is a terrible lesson we already learned elsewhere.

File: cfgov/jinja2/v1/_layouts/base.html [hunk 2]
```diff
@@ -200,6 +200,7 @@
     {# TODO: Remove Else conditional when this is fully moved to wagtail. #}
     {# Check and include page-level JavaScript. #}
     {% if page and page.media %}
+        /* {{ page.media }} */
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_layouts/base.html]: Remove temp code here

File: cfgov/v1/atomic_elements/organisms.py [hunk 3]
```diff
@@ -163,14 +163,42 @@ class Meta:
...
+
+    class Media:
+        js = ["expandable.js"]
+
+
+class Expandable(BaseExpandable):
+    content = blocks.StreamBlock(
+        [
+            ('paragraph', blocks.RichTextBlock(required=False)),
+            ('Well', Well()),
```
Comment (jimmynotjim) [inline cfgov/v1/atomic_elements/organisms.py]: Should `'Well'` be capitalized?

**Comment by jimmynotjim** [comment]
@Scotchester Looks like the organism styles are overwriting the molecule styles, since they come farther down the stack. See https://github.com/cfpb/cfgov-refresh/pull/2157#discussion_r65422134 about mixing atomic types, it never ends well.

File: cfgov/jinja2/v1/_includes/organisms/header.html (hunk 1 above)
Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/header.html]: Oooh, good call.

File: cfgov/jinja2/v1/_layouts/base.html (hunk 2 above)
Comment (Scotchester) [inline cfgov/jinja2/v1/_layouts/base.html]: Got it.

File: cfgov/v1/atomic_elements/organisms.py (hunk 3 above)
Comment (Scotchester) [inline cfgov/v1/atomic_elements/organisms.py]: Hmm, perhaps not.

**Comment by Scotchester** [comment]
@jimmynotjim 

Hmm, yes, I see what you mean. I'm unsure what to do here. Do you think that this means trying to promote Expandables was a mistake and we should abandon this PR (and presumably go with the idea of demoting Wells to a molecule)? Should we just make the global banner styles more specific?

Should we abandon this PR and devise a _new,_ non-Well molecule of some kind to provide a highlighted area of some kind in Expandables?

(Reminder: We'd reeeeally like to have this in time for the Auto Loans launch on Wednesday.)

**Comment by jimmynotjim** [comment]
I think we're ok promoting Expandables, we just need to untangle the banner from them. That probably means promoting the banner to an organism as well, or even making it a modifier to the expandable, Ex. `.o-expandable_target__global-banner`. Thinking it through, that's what we probably should have done from the beginning, even as a molecule. Mixing two atomic components in one element just leads to unexpected consequences when changing one of those components.

**Comment by Scotchester** [comment]
@jimmynotjim Pushed an update to fix the global banner issue. This isn't quite the atomic overhaul you are seeking, but

1. I noticed that we already had `.m-global-banner .o-expandable_target` rules, so simply moving most of the `.m-global-banner_btn` rules into that selector fixed the specificity problem.
2. I also noticed that someone (@anselmbradford, I presume) left a TODO in there to "Refactor to use flyout menu behavior and not include an expandable, so that these style overrides aren't necessary" and I figured any work I did to refactor it would probably soon get reworked to accomplish that TODO.

What do you think?

**Comment by jimmynotjim** [comment]
That works. @KimberlyMunoz could you take a look so this is ready to go for the Auto Loan release next week?

**Comment by KimberlyMunoz** [comment]
I ran migrations and got some errors on the secondary nav. 
image

**Comment by Scotchester** [comment]
@KimberlyMunoz 

Fixed the conflict, which was just in the CHANGELOG. Shouldn't be any issue with migrations; this adds `0088` and there were no other `0088`s. 

Did you run `gulp styles && gulp scripts``?

**Comment by KimberlyMunoz** [comment]
Yep. I ran `./setup.sh`

**Comment by Scotchester** [comment]
Hmm... can't reproduce. I did a `frontend.sh` only, because I was worried the back-end half would blow away my test pages, but it's all working well for me.

- Are there any JS errors in the console?
- Tried disabling the cache or restarting your local server?

File: cfgov/v1/atomic_elements/organisms.py (hunk 3 above)
Comment (kurtw) [inline cfgov/v1/atomic_elements/organisms.py]: probably best to lowercase.

**Comment by Scotchester** [comment]
Conflicts again. This .gitattributes thing isn't working out as well as intended...

**Comment by jimmynotjim** [comment]
Don't worry about changelog conflicts till you're ready to merge. 

I can't duplicate @KimberlyMunoz's issue, everything is working as expected so 👍 from me. Can you give it another shot Kim? @anselmbradford can you test this as well and see if you see the issue above?

**Comment by Scotchester** [comment]
@jimmynotjim Actually, he did try it yesterday. We figured out that my "fix" (mentioned in the Notes section of the OP) wasn't actually a fix, and more work was needed. Consulted with @kurtw, who is going to PR to this branch something that should fix it.

**Comment by Scotchester** [comment]
Updated with changes from #2171, and reverted the change described in the note in the OP. Can I get another quick review, @anselmbradford?

**Comment by anselmbradford** [comment]
👍  @Scotchester you know this already, but please document that this introduces that standalone expandables should not be added in wagtail before an expandablegroup or a JS error will be thrown on the published page.

**Comment by Scotchester** [comment]
Roger; wilco.

Going to fix merge conflicts and get this show on the road!

**Comment by Scotchester** [comment]
@anselmbradford Commented on DFJR-994.

**Pull request merged**